packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -952,20 +953,30 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
             throw error;
         }
 
-        const shouldRetryRequest = !request.noRetry && request.retryCount < this.maxRequestRetries && !(error instanceof NonRetryableError);
+        const isUserRequestedRetry = error instanceof RetryRequest;
+
+        const shouldRetryRequest = isUserRequestedRetry
+            || (!request.noRetry && request.retryCount < this.maxRequestRetries && !(error instanceof NonRetryableError));
+
         if (shouldRetryRequest) {
             request.retryCount++;
 
-            await this._tagUserHandlerError(() => this.errorHandler?.({ ...crawlingContext, error }, error));
+            await this._tagUserHandlerError(() => this.errorHandler?.(this._augmentContextWithDeprecatedError(crawlingContext, error), error));
 
             const { url, retryCount, id } = request;
-            // We don't want to see the stack trace in the logs by default, when we are going to retry the request.
-            // Thus, we print the full stack trace only when CRAWLEE_VERBOSE_LOG environment variable is set to true.
-            const message = process.env.CRAWLEE_VERBOSE_LOG ? error.stack : error;
-            this.log.warning(
-                `Reclaiming failed request back to the list or queue. ${message}`,
-                { id, url, retryCount },
-            );
+
+            if (isUserRequestedRetry) {
+                // Let the user know the request they want retried was rescheduled, while respecting the maximum retry count
+                this.log.info("Rescheduled request for a retry at the user's request", { id, url, retryCount });
+            } else {
+                // We don't want to see the stack trace in the logs by default, when we are going to retry the request.
+                // Thus, we print the full stack trace only when CRAWLEE_VERBOSE_LOG environment variable is set to true.
+                const message = process.env.CRAWLEE_VERBOSE_LOG ? (error as any).stack ?? (error as any).message ?? error : error;
+                this.log.warning(
+                    `Reclaiming failed request back to the list or queue. ${message}`,
+                    { id, url, retryCount },
+                );
+            }
```

Comment:
Why is the logic for the user requested retry and auto-retry different? Maybe instead of having this logic here, we could just wrap the unknown error with the retryable error instance and that would take care of it?

I mean, it's not entirely obvious from this:

- how the user requested retry will behave in terms of VERBOSE_LOG
- is it gonna print stack?

Reply:
If a user throws a RetryRequest (or whatever it will be named to if the naming will change), the only difference is that the user will get a different logging message (the one on top vs the one below for all the other errors)

Reply:
So with the retry request error they will never get a stack and with the other errors they will get the stack under some conditions? I think we could unify the behavior. But I don't have a strong opinion about it.

Reply:
With the retry request there will never be a stack, otherwise if VERBOSE_LOG is set we will log the stack (or message or the error, depending on if those properties exist or not), otherwise we'll just log the error. I can see why that could be confusing so I can add some comments to clarify it!

Reply:
I think we dont need specialing handling, would keep the old code. How does it behave if we keep the old version? It would print the `Reclaiming failed request back to the list or queue.` with the user error message appended, right? That seems good to me, I dont mind its a warning.